aiida/manage/configuration/config.py

```diff
@@ -18,12 +18,34 @@
 
 from aiida.common import json
 
-from .migrations import CURRENT_CONFIG_VERSION, OLDEST_COMPATIBLE_CONFIG_VERSION
+from .migrations import CURRENT_CONFIG_VERSION, OLDEST_COMPATIBLE_CONFIG_VERSION, get_current_version
 from .options import get_option, parse_option, NO_DEFAULT
 from .profile import Profile
 from .settings import DEFAULT_UMASK, DEFAULT_CONFIG_INDENT_SIZE
 
-__all__ = ('Config',)
+__all__ = ('Config', 'create_backup')
+
+
+def create_backup(config_dictionary, filepath):
+    """Create backup of configuration.
+
+    :param config_dictionary: contents of the configuration file
+    :param filepath: absolute path to the current configuration file
+    """
+    from aiida.common import timezone
+    from aiida.cmdline.utils import echo
+
+    version = get_current_version(config_dictionary)
+    filepath_backup = '{}.v{}.{}'.format(filepath, version, timezone.now().strftime('%Y%m%d-%H%M%S'))
+
+    echo.echo_warning('current configuration file `{}` is outdated and will be migrated'.format(filepath))
```

Comment:
To me, these two echo statements don't belong in this function - the function should just create the backup, right?
I would move them to where you do the migration

Reply:
I have looked at the code a bit better and think I have found a solution that is a lot better and aligns with your suggestion. Construction of `Config` from file is now done through `Config.from_file` class method which is now responsible for migrating and creating of backup. This bundles these two operations more closely, and for the better I feel.